### PR TITLE
Add org.zdoom.Raze

### PR DESCRIPTION
**Pull request by smirnov**
### Please confirm your submission meets all the criteria

Build of Raze (source port for Build-engine games) based on GZDoom tech and EDuke32, NBlood and other source ports, using GZDoom build instructions

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **https://github.com/ZDoom/Raze/issues/403**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.zdoom.Raze

File: org.zdoom.Raze.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+- /app/lib/*.a
+- /app/lib/*.la
+- /app/lib/pkgconfig
+
+modules:
+
+- shared-modules/linux-audio/libinstpatch.json
+- shared-modules/linux-audio/fluidsynth2.json
+
+- name: libsndfile
```
Comment (hfiguiere) [inline org.zdoom.Raze.yaml:36]:
```
$ pkg-config --modversion sndfile
1.2.0
```
This is in the runtime

File: org.zdoom.Raze.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  - "-DSYSTEMINSTALL=1"
+  cleanup:
+  - "/lib/*.a"
+  - "/lib/*.la"
+  - "/lib/pkgconfig"
+  - "/include"
+  sources:
+  - type: git
+    url: https://github.com/ZDoom/Raze.git
+    tag: 1.9
```
Comment (hfiguiere) [inline org.zdoom.Raze.yaml:90]:
```suggestion
    tag: '1.9'
```

File: org.zdoom.Raze.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+- --device=all
+
+# Simpler way of preventing KDE related errors
+- --env=KDE_FULL_SESSION=false
+
+cleanup:
+- /app/include
+- /app/lib/*.a
+- /app/lib/*.la
+- /app/lib/pkgconfig
```
Comment (hfiguiere) [inline org.zdoom.Raze.yaml:29]:
```suggestion
- /include
- /lib/*.a
- /lib/*.la
- /lib/pkgconfig
```

File: org.zdoom.Raze.yaml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  - type: file
+    path: org.zdoom.Raze.appdata.xml
+  - type: file
+    path: org.zdoom.Raze.32.png
+  - type: file
+    path: org.zdoom.Raze.64.png
+  - type: file
+    path: org.zdoom.Raze.128.png
+  - type: file
+    path: org.zdoom.Raze.256.png
```
Comment (hfiguiere) [inline org.zdoom.Raze.yaml:113]: remove the smaller ones and keep this one since they are not upstream.

File: org.zdoom.Raze.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  - "/include"
+  sources:
+  - type: git
+    url: https://github.com/ZDoom/Raze.git
+    tag: 1.9
+    commit: 9af265b5ea1ab2872368ae58ad3b5490fab658f1
+  post-install:
+  - install -Dm 644 soundfont/raze.sf2 /app/share/games/raze/soundfonts/raze.sf2
+
+- name: integration
```
Comment (hfiguiere) [inline org.zdoom.Raze.yaml:95]: this doesn't need to be a separate module.

File: org.zdoom.Raze.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2021 Hugo Locurcio -->
+
+<component type="desktop">
+  <id>org.zdoom.Raze</id>
+  <name>Raze</name>
+  <summary>Classic first-person-shooter engine for all Build engine games</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <!-- The Build license is not an open source license as it disallows commercial use. -->
+  <project_license>proprietary</project_license>
```
Comment (bbhtt) [inline org.zdoom.Raze.appdata.xml:10]: https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#license

> Proprietary applications/custom licenses can use LicenseRef-proprietary with a link with a link to the license

File: org.zdoom.Raze.yaml [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  - type: file
+    path: org.zdoom.Raze.256.png
+  build-commands:
+  - install -D raze.sh /app/bin/raze.sh
+  - install -Dm 644 org.zdoom.Raze.desktop -t /app/share/applications
+  - install -Dm 644 org.zdoom.Raze.appdata.xml -t /app/share/metainfo
+  - install -Dm 644 org.zdoom.Raze.32.png /app/share/icons/hicolor/32x32/apps/org.zdoom.Raze.png
+  - install -Dm 644 org.zdoom.Raze.64.png /app/share/icons/hicolor/64x64/apps/org.zdoom.Raze.png
+  - install -Dm 644 org.zdoom.Raze.128.png /app/share/icons/hicolor/128x128/apps/org.zdoom.Raze.png
+  - install -Dm 644 org.zdoom.Raze.256.png /app/share/icons/hicolor/256x256/apps/org.zdoom.Raze.png
```
Comment (bbhtt) [inline org.zdoom.Raze.yaml:121]: Unless you need smaller icons just install the 256px

File: org.zdoom.Raze.yaml (hunk 5 above)
Comment (smirnov) [reply org.zdoom.Raze.yaml:80]: Not quite sure how to go about this one, I've been pretty much following what GZDoom is doing for it's build, except for adding the whole build into "shared-modules" and only have the launcher built in the application yaml (as GZDoom does it), instead I have it defined as a separate module, as some earlier attempts at flatpaking the app suggested, would appreciate any advice.

Comment (hfiguiere) [reply org.zdoom.Raze.yaml:80]: > Not quite sure how to go about this one, I've been pretty much following what GZDoom is doing for it's build,

Two wrong don't make a right.

File: org.zdoom.Raze.yaml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    commands:
+      - raze +fluid_patchset /app/share/games/raze/soundfonts/raze.sf2 "$@"
+    dest-filename: raze.sh
+  - type: file
+    path: org.zdoom.Raze.desktop
+  - type: file
+    path: org.zdoom.Raze.appdata.xml
+  - type: file
+    path: org.zdoom.Raze.256.png
+  build-commands:
```
Comment (hfiguiere) [inline org.zdoom.Raze.yaml:93]:
```suggestion
  - type: script
    commands:
      - raze +fluid_patchset /app/share/games/raze/soundfonts/raze.sf2 "$@"
    dest-filename: raze.sh
  - type: file
    path: org.zdoom.Raze.desktop
  - type: file
    path: org.zdoom.Raze.appdata.xml
  - type: file
    path: org.zdoom.Raze.256.png
  post-install:
  - install -Dm 644 soundfont/raze.sf2 /app/share/games/raze/soundfonts/raze.sf2

```

File: org.zdoom.Raze.desktop
Comment (hfiguiere) [inline org.zdoom.Raze.desktop:1]: this should be submitted upstream.

File: org.zdoom.Raze.256.png
Comment (hfiguiere) [inline org.zdoom.Raze.256.png:1]: this should be submitted upstream

File: org.zdoom.Raze.yaml [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+# We redirect the original ~/.config/raze
+- --persist=.config/raze
+
+# Simpler way of preventing KDE related errors
+- --env=KDE_FULL_SESSION=false
+
+cleanup:
+- /include
+- /lib/*.a
+- /lib/*.la
```
Comment (bbhtt) [inline org.zdoom.Raze.yaml:26]:
```suggestion
- "*.a"
- "*.la"
```

File: org.zdoom.Raze.yaml [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+- name: raze
+  buildsystem: cmake-ninja
+  config-opts:
+  - "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+  - "-DSYSTEMINSTALL=1"
+  cleanup:
+  - "/lib/*.a"
+  - "/lib/*.la"
+  - "/lib/pkgconfig"
+  - "/include"
```
Comment (bbhtt) [inline org.zdoom.Raze.yaml:71]:
```suggestion
```

top level covers it

File: org.zdoom.Raze.yaml [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  - type: file
+    path: org.zdoom.Raze.desktop
+  - type: file
+    path: org.zdoom.Raze.appdata.xml
+  - type: file
+    path: org.zdoom.Raze.256.png
+  build-commands:
+  - install -D raze.sh /app/bin/raze.sh
+  - install -Dm 644 org.zdoom.Raze.desktop -t /app/share/applications
+  - install -Dm 644 org.zdoom.Raze.256.png /app/share/icons/hicolor/256x256/apps/org.zdoom.Raze.png
```
Comment (bbhtt) [inline org.zdoom.Raze.yaml:96]: Metainfo is not being installed

File: org.zdoom.Raze.yaml (hunk 5 above)
Comment (smirnov) [reply org.zdoom.Raze.yaml:80]: Sorry, didn't mean that as a justification, just that  being new to flatpak packaging I didn't know how to properly include the second step in the one in install, thank you very much for providing a suggestion.

**Comment by bbhtt** [comment]
bot, build org.zdoom.Raze

**Comment by smirnov** [comment]
bot, build org.zdoom.Raze

**Comment by smirnov** [comment]
bot, build org.zdoom.Raze

**Comment by Eonfge** [comment]
Looks good. I'm maintaining the Freedoom and GZDoom packages so they have a lot in common with this project.

As far as feedback goes, you might want to change the .yaml indenting to match the style-guide: https://github.com/flathub/flathub/wiki/YAML-Style-Guide It's no hard requirement though... the buildbot will fix it for you next time there it detects an outdated dependency.

As for the icons and metadata that should be submitted upstream; make an issue and see if they care. For GZDoom, they decided not to include it (https://github.com/flathub/org.zdoom.GZDoom), but with Freedoom we have a healthy collaboration.

**Comment by smirnov** [comment]
Thank you for the feedback, since it's the same dev as GZDoom, I guess the answer's going to be the same, but it's worth to at least try and offer a PR.

**Pull request closed without merging**